### PR TITLE
Adding Product Listing Page

### PR DESCRIPTION
**Pull request by bipin7yadav**
mapped products and  added  filter feature
productListingPage
s

### CODE REVIEW

File: src/contexts/ProductContext.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            try{const response = await axios.get("/api/products");
+            setProduct(response.data.products)
+        }
+        catch (error){
+            console.log(error)
+        }
+        })();
+    },[]);
+
+    const FilterReducer = (state,action) =>{
```
Comment (techieshubham30) [inline src/contexts/ProductContext.js:21]: Suggestion: You can create a separate folder for all your reducer implementations.

File: src/pages/AllProducts/AllProducts.css [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    padding:0 1rem;
+    display: flex;
+    flex-direction: column;
+}
+
+.btn{
+    width: 100%;
+}
+
+/* .pro-img{
```
Comment (techieshubham30) [inline src/pages/AllProducts/AllProducts.css]: Remove these comments if not needed.

**Review by techieshubham30 — APPROVED** [review]

File: src/pages/AllProducts/AllProducts.css (hunk 2 above)
Comment (bipin7yadav) [reply src/pages/AllProducts/AllProducts.css]: done

**Review by someshtiwarii — APPROVED** [review]
UI looks clean. Good to go

**Pull request merged**